pipeline-runner/init.R

```diff
@@ -187,16 +198,30 @@ call_data_processing <- function(task_name, input, pipeline_config) {
     assign("scdata", data, pos = ".GlobalEnv")
 
     # upload plot data result to S3
+    tstart <- Sys.time()
+
     plot_data_keys <- send_plot_data_to_s3(pipeline_config, experiment_id, rest_of_results)
 
     # Uplaod count matrix data
     if(upload_count_matrix) {
-        object_key <- upload_matrix_to_s3(pipeline_config, experiment_id, scdata)
-        message('Count matrix uploaded to ', pipeline_config$processed_bucket, ' with key ',object_key)
+
+        # move scdata to worker (direct for dev only)
+        if (upload_count_matrix & pipeline_config$cluster_env != 'development') {
+            object_key <- upload_matrix_to_s3(pipeline_config, experiment_id, scdata)
+            message('Count matrix uploaded to ', pipeline_config$processed_bucket, ' with key ',object_key)
+
+        } else if (upload_count_matrix) {
+            expid_path <- file.path('/worker/data', experiment_id)
+            dir.create(expid_path, showWarnings = FALSE)
+            saveRDS(scdata, file.path(expid_path, 'r.rds'))
```

Comment:
Iva is concerned that we are diverging from our aws environment. If it helps prevent the timeout it's okay, but we need to make sure that it actually helps avoiding the timeout.

Reply:
Can you test? I don't experience timeouts